syntax = "proto2";

message RequiredMessage{
    required string coder_name = 1;
}

message OptionalMessage{
    optional string coder_name = 1;
}

message RepeatedMessage{
    repeated string coder_names = 1;
}

message OneOfMessage{
    oneof any_one{
        string full_name = 1;
        string first_name = 2;
    }
}